File: app/src/main/java/com/demo/justapp/exchanger/ExchangerApp.java
package com.demo.justapp.exchanger;

import android.app.Application;

import com.demo.justapp.exchanger.di.application.AppInjector;

/**
 * @author Sergey Rodionov
 */
public class ExchangerApp extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
         AppInjector.Companion.createAppComponent(this).inject(this);
    }
}

File: app/src/main/kotlin/com/demo/justapp/exchanger/ExchangerApp.kt
package com.demo.justapp.exchanger

import android.app.Application
import com.demo.justapp.exchanger.di.application.ApplicationInjector
import timber.log.Timber

class ExchangerApp : Application() {

    override fun onCreate() {
        super.onCreate()
        ApplicationInjector.createAppComponent(this)
        setupTimber()
    }

    private fun setupTimber() {
        if (BuildConfig.DEBUG) Timber.plant(Timber.DebugTree())
    }

}